### PR TITLE
Header bar improvement

### PR DESCRIPTION
**Pull request by yxl**
One of my midlet uses the menu commands of `OK` and `BACK` heavily to navigate among different screens. The user needs two clicks to reach these commands, since they are folded in a hidden drawer. To make the midlet easy to use, I move these commands out of the drawser as the Nokia Asha 3110 does.

The `OK` command is placed on right of the header bar and the `BACK` command is placed at the bottom right corner of the midlet.

The drawer will be hidden if it is empty.

I abandon the url parameter of `nativeMenu`, which forces the header bar to show. The header bar is hidden if the midlet runs in full screen mode. Otherwise the header bar shows.

It also fixes two small UI bugs.

It may be easier to review by each commit and run `/index.html?jars=tests/tests.jar&midletClassName=gfx.CanvasWithHeaderTest` to test the header bar.

### CODE REVIEW

File: tests/automation.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ casper.options.onWaitTimeout = function() {
 var gfxTests = [
   { name: "gfx/AlertTest", maxDifferent: 1942 },
   { name: "gfx/CanvasTest", maxDifferent: 0 },
+  { name: "gfx/CanvasWithHeaderTest", maxDifferent: 2000 },
```
Comment (mykmelez) [inline tests/automation.js]: Does this really need to be as high as 2000? I see 1368 different pixels in my runs on Mac, which typically has the highest number of different pixels.

Comment (mykmelez) [inline tests/automation.js]: And the number was 969 on the Travis run.

**Comment by mykmelez** [comment]
Other than that nit, this looks fine, but I'd like @marco-c to look at it too.

File: tests/automation.js (hunk 1 above)
Comment (yxl) [inline tests/automation.js]: I reduced maxDifferent to 1400 to be compatible with that on Mac. It's value varies a lot with different platform. On my ubuntu desktop the value is 0.

**Comment by marco-c** [comment]
I'll merge this once conflicts are resolved!

**Pull request merged**